### DISCUSSION TITLE
SSH agent support

### DISCUSSION THREAD

**Discussion by allblue** [1Remote/1Remote#481] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, only local key files can be configured, but ssh agent mode is more flexible and can automatically choose among multiple private keys. When there are many sessions, it is more convenient to manage.
目前仅可配置本地秘钥文件，而ssh agent方式更为灵活，且可以在多个私钥间自动选择，当session较多时，管理起来更方便。

**Describe the solution you'd like**
SSH agent is usually implemented by pagent in putty on Windows, and I hope to integrate pagent authentication
ssh agent在windows下一般通过putty中的pagent实现，希望能集成pagent认证

**Describe alternatives you've considered**
no

**Additional context**
no

**Comment by majkinetor** [comment]
I guess agent could be supported, not sure what is blocking it now since I didn't use that feature so far. @VShawn, what do you think?

**Reply by VShawn** [reply]
I have very little understanding of SSH, and I have never used SSH agent before. I am not sure if it is possible to add this feature.

## Append

I searched through Google and couldn't find an example of using SSH agent with Putty.

## Append 2

```
Meanwhile, PuTTY on Windows has pageant.exe for the same purpose. (Other programs, such as WinSCP, also use Pageant.)
``` 

https://superuser.com/questions/878943/how-to-use-putty-for-forwarding-keys

**Reply by allblue** [reply]
## pageant document
1. official: https://the.earth.li/~sgtatham/putty/0.78/htmldoc/Chapter9.html#pageant
2. winscp: https://winscp.net/eng/docs/ui_pageant
3. Using Pageant Putty Agent to Unleash Your SSH Key Use: https://adamtheautomator.com/pageant-putty/

## kitty document
kitty use [Kageant](https://www.9bis.net/kitty/files/kageant.exe)
official: https://www.9bis.net/kitty/?#!pages/KeyConfirmation.md
